File: tejvanta-ui/src/types/Instrument.ts
export interface Instrument {
  id: number
  symbol: string
  name: string
  exchange: string
  tickSize: number
  lotSize: number
  segment: string
  series: string
}

export interface Candle {
  id: number
  instrumentId: number
  timestamp: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  timeframe: string
}

export type DataSourceType = 'simulation'

export type ConnectionState = 'disconnected' | 'connecting' | 'connected' | 'error'

export interface DataConnectionStatus {
  source: DataSourceType
  state: ConnectionState
  message?: string
}

export interface Tick {
  instrumentId: number
  symbol?: string
  timestamp: string
  price: number
  volume: number
  bid: number
  ask: number
  open: number
  high: number
  low: number
  prevClose: number
  change: number
  changePercent: number
}

File: Tejvanta.Api/src/Domain/Entities/Instrument.cs
namespace Tejvanta.Api.Domain.Entities;

public class Instrument
{
    public int Id { get; set; }
    public string Symbol { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Exchange { get; set; } = "NSE";
    public decimal TickSize { get; set; } = 0.05m;
    public int LotSize { get; set; } = 1;
    public string Segment { get; set; } = "EQ";
    public string Series { get; set; } = "EQ";
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
